### ISSUE TITLE
Possible bug in PeriodCollection gaps()

### ISSUE THREAD

**Issue by jonagoldman** [spatie/period#58]
Given two **equal** periods with boundaries:
```php
$collection = new PeriodCollection(
    Period::make('2018-01-01', '2018-01-03', null, Boundaries::EXCLUDE_END),
    Period::make('2018-01-01', '2018-01-03', null, Boundaries::EXCLUDE_END)
);

$gaps = $collection->gaps();
```
I expected gaps to return **0** periods, but I get **1** period instead:

```php
// $gaps->count() == 1
// $period->getStart() == '2018-01-03'
// $period->getEnd() == '2018-01-03'
```

Is this expected?

**Comment by brendt** [comment]
That's a valid bug. I'll look into it this week.

**Comment by brendt** [comment]
Fixed with 1.4.5: https://github.com/spatie/period/releases/tag/1.4.5

It'll probably take a few minutes before packagist knows about this new tag.

**Comment by jonagoldman** [comment]
Wow that was fast! Thanks!
